# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/ActorsParent.cpp
# Commit: a576f5346c6c
# Full Hash: a576f5346c6cfc32a5b147158e48ec1f4fcf2a8f
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2025-06-12 16:31:09
# Regressor Bug: 1962133
# File Overlap Count: 5
# Overlapping Files: dom/quota/DirectoryMetadata.h, dom/quota/DirectoryMetadata.cpp, dom/quota/OriginOperations.cpp, dom/quota/moz.build, dom/quota/ActorsParent.cpp
# Description:
#   Bug 1962133 - QM: Prepare metadata header infrastructure for accessed flag addition and reduce duplication; r=dom-storage-reviewers,hsingh
#   
#   This patch restructures metadata header reading and writing code to reduce
#   duplication and establish clean infrastructure for adding the accessed flag to
#   the metadata file header.
# ==============================================================================

diff -r f4284b0494fc -r a576f5346c6c dom/quota/ActorsParent.cpp
--- a/dom/quota/ActorsParent.cpp	Wed Jun 11 20:01:43 2025 +0000
+++ b/dom/quota/ActorsParent.cpp	Wed Jun 11 20:09:11 2025 +0000
@@ -9,6 +9,7 @@
 // Local includes
 #include "CanonicalQuotaObject.h"
 #include "ClientUsageArray.h"
+#include "DirectoryMetadata.h"
 #include "Flatten.h"
 #include "FirstInitializationAttemptsImpl.h"
 #include "GroupInfo.h"
@@ -3456,15 +3457,8 @@
                  GetBinaryOutputStream(*file, FileFlag::Truncate));
   MOZ_ASSERT(stream);
 
-  QM_TRY(MOZ_TO_RESULT(stream->Write64(aTimestamp)));
-
-  QM_TRY(MOZ_TO_RESULT(stream->WriteBoolean(aPersisted)));
-
-  // Reserved data 1
-  QM_TRY(MOZ_TO_RESULT(stream->Write32(0)));
-
-  // Reserved data 2
-  QM_TRY(MOZ_TO_RESULT(stream->Write32(0)));
+  QM_TRY(MOZ_TO_RESULT(WriteDirectoryMetadataHeader(
+      *stream, OriginStateMetadata{aTimestamp, aPersisted})));
 
   // Legacy field, previously used for suffix. The value is no longer used, but
   // we continue writing the correct suffix value to preserve compatibility
@@ -3524,22 +3518,10 @@
 
   FullOriginMetadata fullOriginMetadata;
 
-  QM_TRY_UNWRAP(fullOriginMetadata.mLastAccessTime,
-                MOZ_TO_RESULT_INVOKE_MEMBER(binaryStream, Read64));
-
-  QM_TRY_UNWRAP(fullOriginMetadata.mPersisted,
-                MOZ_TO_RESULT_INVOKE_MEMBER(binaryStream, ReadBoolean));
-
-  QM_TRY_INSPECT(const bool& reservedData1,
-                 MOZ_TO_RESULT_INVOKE_MEMBER(binaryStream, Read32));
-  if (reservedData1 != 0) {
-    QM_TRY(MOZ_TO_RESULT(false));
-  }
-
-  // XXX Use for the persistence type.
-  QM_TRY_INSPECT(const bool& reservedData2,
-                 MOZ_TO_RESULT_INVOKE_MEMBER(binaryStream, Read32));
-  Unused << reservedData2;
+  QM_TRY_INSPECT(const OriginStateMetadata& originStateMetadata,
+                 ReadDirectoryMetadataHeader(*binaryStream));
+
+  static_cast<OriginStateMetadata&>(fullOriginMetadata) = originStateMetadata;
 
   fullOriginMetadata.mPersistenceType = aPersistenceType;
 